exporter/opentelemetry-exporter-prometheus-remote-write/tests/test_prometheus_remote_write_exporter.py

```diff
@@ -0,0 +1,377 @@
+        expected_timeseries.labels.append(
+            self._exporter.create_label("__name__", "testname")
+        )
+        expected_timeseries.labels.append(
+            self._exporter.create_label("resource_name", "resource_value")
+        )
+        expected_timeseries.labels.append(
+            self._exporter.create_label("record_name", "record_value")
+        )
+        expected_timeseries.samples.append(
+            self._exporter.create_sample(10, 5.0),
+        )
+        timeseries = self._exporter.create_timeseries(
+            export_record, "testname", 5.0,
+        )
+        self.assertEqual(timeseries, expected_timeseries)
+
+
+class ResponseStub:
+    def __init__(self, status_code):
+        self.status_code = status_code
+        self.reason = "dummy_reason"
+        self.content = "dummy_content"
+
+
+class TestExport(unittest.TestCase):
+    # Initializes test data that is reused across tests
+    def setUp(self):
+        self._exporter = PrometheusRemoteWriteMetricsExporter(
+            endpoint="/prom/test_endpoint"
+        )
+
+    # Ensures export is successful with valid export_records and config
+    @mock.patch("requests.post", return_value=ResponseStub(200))
+    def test_export(self, mock_post):
+        test_metric = Counter("testname", "testdesc", "testunit", int, None)
+        labels = {"environment": "testing"}
+        record = ExportRecord(
+            test_metric, labels, SumAggregator(), Resource({}),
+        )
+        result = self._exporter.export([record])
+        self.assertIs(result, MetricsExportResult.SUCCESS)
+        self.assertEqual(mock_post.call_count, 1)
+
+    @mock.patch("requests.post", return_value=ResponseStub(200))
+    def test_valid_send_message(self, mock_post):
+        result = self._exporter.send_message(bytes(), {})
+        self.assertEqual(mock_post.call_count, 1)
+        self.assertEqual(result, MetricsExportResult.SUCCESS)
+
+    @mock.patch("requests.post", return_value=ResponseStub(404))
+    def test_invalid_send_message(self, mock_post):
+        result = self._exporter.send_message(bytes(), {})
+        self.assertEqual(mock_post.call_count, 1)
+        self.assertEqual(result, MetricsExportResult.FAILURE)
+
+    # Verifies that build_message calls snappy.compress and returns SerializedString
+    @mock.patch("snappy.compress", return_value=bytes())
+    def test_build_message(self, mock_compress):
+        test_timeseries = [
+            TimeSeries(),
+            TimeSeries(),
+        ]
+        message = self._exporter.build_message(test_timeseries)
+        self.assertEqual(mock_compress.call_count, 1)
+        self.assertIsInstance(message, bytes)
+
+    # Ensure correct headers are added when valid config is provided
+    def test_get_headers(self):
+        self._exporter.headers = {"Custom Header": "test_header"}
+
+        headers = self._exporter.get_headers()
+        self.assertEqual(headers.get("Content-Encoding", ""), "snappy")
```

Comment:
This can be misleading if the test case fails. If `"Content-Encoding"` was supposed to be in the keys of `header`, then you should not artificially hide that fact under a different exception. If `"Content-Encoding"` is not in the keys of `header`, then you should be raising a `KeyError` and not an `AssertionError`. This is because if the test case fails, the current implementation of this assertion does not make it clear if `"Content-Encoding"` was not in the keys of `header` or if its original value was `""`.

Reply:
Good point, fixed. I did not realize the standard function errors do not break the test.

Reply:
Any exception raised in a test will make it fail. What we don't want is to introduce ambiguity in a test case failure because the same exception (`AssertionError`) would be raised if the key was missing from `headers` or if its value was `""`. Both are wrong but should raise different exceptions to clearly indicate what went wront.